### DISCUSSION TITLE
URL encoding query parameters

### DISCUSSION THREAD

**Discussion by andrewbaxter** [Byron/google-apis-rs#468] [Q&A]
In storage, if you try to send a request for object `a/b` it fails with a 404.  Manually url-encoding the object key before making the request solves the issue.

I see in the code it's using `Params` which has a flag to optionally encode parameters. This is disabled for "object" in the storage library, but it doesn't look like the list of escaped characters includes `/` even if it were enabled.

Do you recall the history of decisions regarding url encoding? My naive assumption would be to always url encode all parameters using something like [urlencoding](https://docs.rs/urlencoding/latest/urlencoding/).  If there's nothing blocking it I'd be happy to make an MR.

**Comment by Byron** [comment]
Thanks for bringing this up.

I don't recall what led to the decision or if the flag is controlled by the API itself.

Making changes that are too general can cause a lot of problems for other APIs and there are no tests to catch this. The last time I was open to such general changes it caused more trouble than it solved.

Hence, a MR is definitely welcome if it can be tied to a per-API override, which then can be set for the respective API, `storage` in this case. The youtube-API might have an override set as example, which could be a template for implementing another one.
